internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -351,6 +380,14 @@ func resourceOrchestratedVirtualMachineScaleSetCreate(d *pluginsdk.ResourceData,
 		props.Sku = sku
 	}
 
+	if v, ok := d.GetOk("sku_profile"); ok {
+		if props.Sku == nil || pointer.From(props.Sku.Name) != "Mix" {
```

Comment:
Could this be implemented as a customiseDiff to catch this configuration issue at plan time rather than erroring out here?

Reply:
moved the logic into `customiseDiff `. Thanks.